pandas/tests/groupby/test_groupby.py

```diff
@@ -3245,3 +3245,14 @@ def test_get_group_axis_1():
         }
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_ffill_with_duplicated_index():
+    # GH#43412
+    df = DataFrame({"a": [1, 2, np.nan, 3, 4, 5]}, index=[0, 1, 2, 0, 1, 2])
+
+    result = df.groupby(level=0).shift().ffill()
```

Comment:
I think the point of the original issue was to test `groupby.ffill`

Reply:
Sure!

I'm testing this and when I use `groupby(level=0)` and `ffill()` methods of dataframe, it gives me an error of different types:

```python
def test_groupby_ffill_with_duplicated_index():
  # GH#43412
  df = DataFrame({"a": [1, 2, 3, 4, np.nan, np.nan]}, index=[0, 1, 2, 0, 1, 2])
  
  result = df.groupby(level=0).ffill()
  expected = DataFrame(
    {"a": [1, 2, 3, 4, 2, 3]}, index=[0, 1, 2, 0, 1, 2]
  )
 ```
```console
>       tm.assert_frame_equal(result, expected)
E       AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="a") are different
E       
E       Attribute "dtype" are different
E       [left]:  float64
E       [right]: int64
```

This behaviour is expected?

Anyways, I will write test with a float64 type dataframe.